botocore/docs/example.py

```diff
@@ -48,6 +50,8 @@ def document_recursive_shape(self, section, shape, **kwargs):
     def document_shape_default(self, section, shape, history, include=None,
                                exclude=None, **kwargs):
         py_type = py_default(shape.type_name)
+        if self._context.get('streaming_shape') == shape:
```

Comment:
You are right. It is only for documentation, and we won't call it too often. I have removed the cache in later commit.